iml-gui/crate/src/components/command_modal.rs

```diff
@@ -551,21 +551,21 @@ where
     match Request::api_query(T::endpoint_name(), &ids) {
         Ok(req) => req.fetch_json_data(data_to_msg).await,
         Err(_) => {
-            // we always can url encode a vector of u32-s
+            // we always can url encode a vector of i32-s
             unreachable!("Cannot encode request for {} with params {:?}", T::endpoint_name(), ids)
         }
     }
 }
 
-fn extract_uri_id<T: EndpointName>(input: &str) -> Option<u32> {
+fn extract_uri_id<T: EndpointName>(input: &str) -> Option<i32> {
     lazy_static::lazy_static! {
         static ref RE: Regex = Regex::new(r"/api/(\w+)/(\d+)/").unwrap();
```

Comment:
```suggestion
        static ref RE: Regex = Regex::new(r"/api/(\w+)/(-?\d+)/").unwrap();
```
The rationale is that since now the ids can be negative in principle, the urls can become like `/api/command/-12/`.
This however might be enforced somewhere else, so the flexibility might not be required.

Reply:
The invariant has been loosened, but in principle we won't accept negative numbers. I don't think we should change this to account for negatives.

Reply:
in other words, the ids should always be > 0.

Reply:
Ok, then.